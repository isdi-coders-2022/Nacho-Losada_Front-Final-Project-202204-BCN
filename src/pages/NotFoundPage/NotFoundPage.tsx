import styled from "styled-components";

const NotFoundPagePageStyled = styled.div`
  height: 60vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;

  img {
    width: 200px;
  }

  p {
    max-width: 300px;
    background: linear-gradient(90deg, #e6a94d 50%, #dfd1b2 100%);
    -webkit-background-clip: text;
    background-clip: text;
    -webkit-text-fill-color: transparent;
    font-size: 2rem;
    text-shadow: 3px 3px 1px rgba(230, 169, 77, 0.25);
  }

  .container {
    display: flex;
    flex-direction: column;

    align-items: center;
    width: fit-content;
  }
`;

const NotFoundPage = (): JSX.Element => {
  return (
    <NotFoundPagePageStyled>
      <div className="container">
        <img
          src="images/poros/notfound-poro.webp"
          alt="poro chafado pensando en porogalletas"
        />

        <p>Ooops!! Parece que la p√°gina que buscas no existe :(</p>
      </div>
    </NotFoundPagePageStyled>
  );
};

export default NotFoundPage;
