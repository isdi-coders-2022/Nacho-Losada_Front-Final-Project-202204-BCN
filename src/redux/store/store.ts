import { configureStore } from "@reduxjs/toolkit";
import ownSummonersReducer from "../features/ownSummonersSlice";
import summonersReducer from "../features/summonersSlice";
import uiReducer from "../features/uiSlice";
import userReducer from "../features/userSlice";

export const store = configureStore({
  reducer: {
    user: userReducer,
    ui: uiReducer,
    summoners: summonersReducer,
    ownSummoners: ownSummonersReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
