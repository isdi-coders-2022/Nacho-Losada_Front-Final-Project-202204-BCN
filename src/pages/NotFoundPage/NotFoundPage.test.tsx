import NotFoundPage from "./NotFoundPage";
import TestRenderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../redux/store/store";

describe("Given a NotFoundPage", () => {
  describe("When it's rendered", () => {
    test("Then it should match this snapshot", () => {
      const renderedNotFoundPage = TestRenderer.create(
        <Provider store={store}>
          <NotFoundPage />
        </Provider>
      );

      expect(renderedNotFoundPage).toMatchSnapshot();
    });
  });
});
