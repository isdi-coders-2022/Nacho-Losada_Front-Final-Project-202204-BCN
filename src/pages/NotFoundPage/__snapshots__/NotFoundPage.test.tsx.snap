// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a NotFoundPage When it's rendered Then it should match this snapshot 1`] = `
<div
  className="sc-bczRLJ gQRBs"
>
  <div
    className="container"
  >
    <img
      alt="poro chafado pensando en porogalletas"
      src="images/poros/notfound-poro.webp"
    />
    <p>
      Ooops!! Parece que la p√°gina que buscas no existe :(
    </p>
  </div>
</div>
`;
